shared/utility/io/uart.cpp

```diff
@@ -140,6 +137,7 @@ namespace utility::io {
             }
 
             // Set the speed flags to "Custom Speed" (clear the existing speed, and set the custom speed flags)
+            // TODO(KipHamiltons) Pretty sure this is undefined behaviour
```

Comment:
This is entirely defined and OK behaviour. This is a standard C way of clearing a flag
```suggestion
```

Reply:
ahh shit. I was misreading this as a minus not tilde.
It still warns that it's implementation defined though

Reply:
It does? Maybe there is something strange we are doing then...

Reply:
Pretty sure it was a narrowing conversion in the implicit cast before the bitwise `and`. I [checked on compiler explorer](https://godbolt.org/z/PrE5eh71v) and both gcc and clang treat narrowing into an int like this as we would expect